lib/datadog/core/environment/variable_helpers.rb

```diff
@@ -24,10 +24,34 @@ def env_to_float(var, default = nil)
           var && ENV.key?(var) ? ENV[var].to_f : default
         end
 
-        def env_to_list(var, default = [])
+        # Parses comma- or space-separated lists.
+        #
+        # If a comma is present, then the list is considered comma-separated.
+        # Otherwise, it is considered space-separated.
+        #
+        # After the entries are separated, commas and whitespaces that are
+        # either trailing or leading are trimmed.
+        #
+        # Empty entries, after trimmed, are also removed from the result.
+        def env_to_list(var, default = [], comma_separated_only: false)
           var = decode_array(var)
           if var && ENV.key?(var)
-            ENV[var].split(',').map(&:strip)
+            value = ENV[var]
+
+            values = if value.include?(',') || comma_separated_only
+                       value.split(',')
+                     else
+                       value.split # space splitting is the default
```

Comment:
Minor: In the spirit of self-documenting code, 

```suggestion
                       value.split(' ')
```

is faster to read than reading the comment ;)

Reply:
Rubocop yelled at me for having the parameter that matches the default, so I removed it.
I'll add it back.

Reply:
Ahh Rubocop... That extremely opinionated colleague that we know nothing about :)